smack-extensions/src/main/java/org/jivesoftware/smackx/muc/packet/GroupChatInvitation.java

```diff
@@ -81,22 +71,85 @@
      * <code>chat.example.com</code>.
      *
      * @param roomAddress the address of the group chat room.
+     * @deprecated use {@link #GroupChatInvitation(EntityBareJid)} instead.
      */
+    @Deprecated
     public GroupChatInvitation(String roomAddress) {
-        this.roomAddress = roomAddress;
+        this(JidCreate.entityBareFromOrThrowUnchecked(roomAddress));
+    }
+
+    public GroupChatInvitation(EntityBareJid roomAddress) {
+        this(roomAddress, null, null, false, null);
+    }
+
+    public GroupChatInvitation(EntityBareJid mucJid, String password, String reason, boolean _continue, String thread) {
+        this.roomAddress = Objects.requireNonNull(mucJid);
+        this.password = password;
+        this.reason = reason;
+        this._continue = _continue;
+        this.thread = thread;
     }
 
     /**
      * Returns the address of the group chat room. GroupChat room addresses
      * are in the form <code>room@service</code>, where <code>service</code> is
      * the name of group chat server, such as <code>chat.example.com</code>.
      *
+     * @deprecated use {@link #getRoomAddressJid()} instead.
```

Comment:
If we deprecate a method, then please add also a "TODO: Remove in Smack X.Y" comment, so that the method is eventually removed.